tracer/src/Datadog.Trace/Iast/InsecureAuthAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        try
+        {
+            // Check for Authorization header
+            const string headerName = HeaderNames.Authorization;
+            if (!headers.TryGetValue(headerName, out var authHeader))
+            {
+                return;
+            }
+
+            if (IgnoredStatusCode(statusCode))
+            {
+                return;
+            }
+
+            var detectedScheme = InsecureSchemeDetected(authHeader);
+            if (detectedScheme is null)
+            {
+                return;
+            }
+
+            IastModule.OnInsecureAuthProtocol($"{headerName}: {detectedScheme}", integrationId);
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(InsecureAuthAnalyzer)}.{nameof(AnalyzeInsecureAuth)}.netcore exception");
+        }
+    }
+#endif
+
+#if NETFRAMEWORK
+    public static void AnalyzeInsecureAuth(System.Collections.Specialized.NameValueCollection headers, IntegrationId integrationId, int statusCode)
+    {
+        try
+        {
+            // Check for Authorization header
+            const string headerName = "Authorization";
+            var authHeader = headers[headerName];
+            if (string.IsNullOrEmpty(authHeader))
+            {
+                return;
+            }
+
+            if (IgnoredStatusCode(statusCode))
+            {
+                return;
+            }
+
+            var detectedScheme = InsecureSchemeDetected(authHeader);
+            if (detectedScheme is null)
+            {
+                return;
+            }
+
+            IastModule.OnInsecureAuthProtocol($"{headerName}: {detectedScheme}", integrationId);
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(InsecureAuthAnalyzer)}.{nameof(AnalyzeInsecureAuth)}.net461 exception");
+        }
+    }
+#endif
+
+    private static bool IgnoredStatusCode(int statusCode)
+    {
+        // To minimize false positives when we get auth credentials to a page that doesn't exist
+        // (e.g. happens with vulnerability scanners), we'll just ignore this vulnerability when
+        // there is no success response.
+        return statusCode >= 400;
+    }
+
+    private static string? InsecureSchemeDetected(string authHeader)
+    {
+        var insecureSchemes = new[] { "Basic", "Digest" };
+
+        // The auth header can be a concatenation of multiple Authorization headers (concatenated by a comma)
+        // This is not a standard practice for the Authorization header but can theoretically happen.
+        var elements = authHeader.Split(',');
+        foreach (var element in elements)
+        {
+            var scheme = element.Trim(' ').Split(' ')[0];
+            var detectedScheme = insecureSchemes.FirstOrDefault(s => s.Equals(scheme, StringComparison.OrdinalIgnoreCase));
+            if (detectedScheme is not null)
+            {
+                return detectedScheme;
+            }
```

Comment:
Given there's only 2 schemes, we can probably avoid the Linq allocations here by doing it manually? It's a bit uglier but meh?
```suggestion
            if ("Basic".Equals(scheme, StringComparison.OrdinalIgnoreCase)
            {
                return "Basic";
            }

            if ("Digest".Equals(scheme, StringComparison.OrdinalIgnoreCase)
            {
                return "Digest";
            }
```

Reply:
fixed in b1e18bd